src/audio/copier.c

```diff
@@ -202,38 +208,97 @@ static struct comp_dev *create_host(struct comp_dev *parent_dev, struct copier_d
 	return dev;
 }
 
+static int init_dai(struct comp_dev *parent_dev, struct copier_data *cd,
+		    const struct comp_driver *drv,
+		    struct comp_ipc_config *config,
+		    struct ipc4_copier_module_cfg *copier,
+		    struct pipeline *pipeline,
+		    struct ipc_config_dai *dai,
+		    enum ipc4_gateway_type type)
+{
+	struct comp_dev *dev;
+	int ret;
+
+	create_endpoint_buffer(parent_dev, cd, config, copier);
+
+	dev = drv->ops.create(drv, config, dai);
+	if (!dev)
+		return -EINVAL;
+
+	if (dai->direction == SOF_IPC_STREAM_PLAYBACK)
+		pipeline->sink_comp = dev;
+	else
+		pipeline->source_comp = dev;
+
+	pipeline->sched_id = config->id;
+
+	list_init(&dev->bsource_list);
+	list_init(&dev->bsink_list);
+
+	ret = comp_dai_config(dev, dai, copier);
+	if (ret < 0)
+		return ret;
+
+	if (dai->direction == SOF_IPC_STREAM_PLAYBACK) {
+		comp_buffer_connect(dev, config->core, cd->endpoint_buffer[cd->endpoint_num],
+				    PPL_CONN_DIR_BUFFER_TO_COMP);
+		cd->bsource_buffer = true;
+	} else {
+		comp_buffer_connect(dev, config->core, cd->endpoint_buffer[cd->endpoint_num],
+				    PPL_CONN_DIR_COMP_TO_BUFFER);
+		cd->bsource_buffer = false;
+	}
+
+	cd->converter[0] = get_converter_func(&copier->base.audio_fmt, &copier->out_fmt, type,
```

Comment:
Why always converter 0. When would we need to use converter 1 ?

Reply:
For copier with gateway support, we only use converter 0 since there is only one sink buffer.
In the case of simple copier, there are max 4 sink buffer.  I will define a macro to make it clear